source/common/filter/tcp_proxy.h

```diff
@@ -30,7 +30,11 @@ namespace Filter {
   COUNTER(downstream_cx_tx_bytes_total)                                                            \
   GAUGE  (downstream_cx_tx_bytes_buffered)                                                         \
   COUNTER(downstream_cx_total)                                                                     \
-  COUNTER(downstream_cx_no_route)
+  COUNTER(downstream_cx_no_route)                                                                  \
+  COUNTER(upstream_pause_reading)                                                                  \
```

Comment:
Is it reasonable to put upstream/downstream in the filter stats and upstream-pauses (and resumes) in the upstream cluster stats?  Adding downstream-pauses in the upstream stats seems odd to me but maybe that's me not having fully absorbed what goes where in envoy yet :-)